### PR TITLE
Fix severe 100% CPU bug.

### PR DESCRIPTION
**Pull request by brstgt**

### CODE REVIEW

File: src/JobInterface.php [hunk 1]
```diff
@@ -64,4 +64,21 @@ public function sendData($data);
...
+    public function complete($data = null);
+
+    /**
+     * Fails the job with optional error message
+     *
+     * @param null $error
+     * @return mixed
+     */
+    public function fail($error = null);
+
```
Comment (bzikarsky) [inline src/JobInterface.php]: image

The CS check doesn't like that last newline before the closing brace.  ;-)

**Comment by brstgt** [comment]
But hopefully now :)

**Comment by bzikarsky** [comment]
I was kinda confused for a second, because you pushed all the fixes into a single PR.

Can you do me a favour and rebasing the fixes (missing interface methods and buffer-management-fix) into 2 separate branches, so they can be handled indepedently.

Regarding the CPU hog bug: There is a test-failure (NULL dereference). I didn't have time to look into it in detail, but maybe you can fix this. :smile:

Thanks for all your contributions -- Are you planning to use react-gearman in production?

**Comment by brstgt** [comment]
Check your email ;)

**Pull request closed without merging**